modules/@apostrophecms/doc/index.js

```diff
@@ -221,32 +221,22 @@ module.exports = {
       '@apostrophecms/doc-type:afterDelete': {
         // If deleting draft also delete published, but
         // not vice versa ("undo publish" is a thing).
-        async deleteOtherModeAfterDelete(req, doc) {
+        async deleteOtherModeAfterDelete(req, doc, options) {
           if (doc.aposLocale.endsWith(':draft')) {
             return cleanup('published');
           }
+          if (doc.aposLocale.endsWith(':published')) {
+            return cleanup('previous');
+          }
           async function cleanup(mode) {
```

Comment:
This no longer uses `mode`, so I'm not sure if passing `'previous'` in above is getting you what you wan.